components/header.js

```diff
@@ -1,9 +1,53 @@
 import styled from "styled-components";
+import { useRouter } from "next/router";
+import Router from "next/router";
+import backButton from "../public/back_button.svg";
+import Image from "next/image";
 
 export default function Header() {
+  const { pathname } = useRouter();
+
+  function printSubhead(pathname) {
+    switch (pathname) {
+      case "/":
+        return "Dashboard";
+      case "/expenses":
+        return "Ausgaben";
+      case "/expenses/add":
+        return "Ausgabe hinzufügen";
+      case "/expenses/details/":
+        return "Ausgabendetails";
+      case "/revenues":
+        return "Einnahmen";
+      case "/profile":
+        return "Profil";
+    }
+
+    if (pathname.includes("/expenses/details/")) {
+      return "Ausgabendetails";
+    }
+  }
```

Comment:
this could be a `util` function. living in a `util` folder and imported here. So you don't have all the code noise in the component and extracting just the string you need.